src/pip/_internal/models/search_scope.py

```diff
@@ -77,11 +77,26 @@ def __init__(
     def get_formatted_locations(self):
         # type: () -> str
         lines = []
+        index_urls = []
         if self.index_urls and self.index_urls != [PyPI.simple_url]:
-            lines.append(
-                'Looking in indexes: {}'.format(', '.join(
-                    redact_auth_from_url(url) for url in self.index_urls))
-            )
+            for url in self.index_urls:
+
+                # Parse the URL
+                purl = urllib_parse.urlsplit(url)
+
+                # URL is generally invalid if scheme and netlock is missing
+                # there are issues with Python and URL parsing, so this test
+                # is a bit crude. See bpo-20271, bpo-23505. Python doesn't
+                # always parse invalid URLs correctly - it should raise
+                # exceptions for malformed URLs
+                if not purl.scheme and not purl.netloc:
+                    logger.warning('index-url {} is invalid.'.format(url))
```

Comment:
This should show the redacted URL as well, not the original.

Reply:
Added redacted URL to warning.